SCAutolib/models/softhsm2_conf.py

```diff
@@ -0,0 +1,67 @@
+"""
+This module provide information and methods to create and modify
+softhsm2.conf file. Following methods are implemented:
+
+create: create content of internal file object representing softhsm2.conf
+set:   set or modify content of internal file object representing softhsm2.conf
+save:  save content of internal file object representing softhsm2.conf to
+       softhsm2.conf file
+clean: removes softhsm2.conf file
+"""
+
+from pathlib import Path
+
+from SCAutolib import logger, TEMPLATES_DIR
+from SCAutolib.models.file import File
+
+
+class SoftHSM2Conf(File):
+    """
+    This class contains information and methods to handle softhsm2.conf file
+    """
+    _template = Path(TEMPLATES_DIR, "softhsm2.conf")
+    _conf_file = None
+    _content = None
+    _card_dir = None
+
+    def __init__(self, filepath: str, card_dir: str):
+        """
+        Init of SoftHSM2Conf
+
+        :param filepath: path where config file should be saved
+        :type filepath: str
+        :param card_dir: parameter to be updated in config file
+        :type card_dir: str
+        """
+        self._conf_file = Path(filepath)
+        self._card_dir = card_dir
+
+    def create(self):
+        """
+        Populate internal file object with content based on template.
+        """
+        with self._template.open('r') as template:
+            content = template.readlines()
+        self._content = []
+        for line in content:
+            modified = line.format(card_dir=self._card_dir)
+            self._content.append(modified)
+
+        logger.info(f"Creating content of {self._conf_file} based on {self._template}")
+        logger.info(f" {self._conf_file}: directories.tokendir needs to be updated")
+
+    def set(self, *args):
+        """
+        Raise NotImplementedError as set method for softHSM2 is not implemented
+        """
+        # parent class set method does not work as softHSM2 conf does not have
+        # sections. Method do modify softHSM2 conf is not implemented
+        logger.warning("softhsm2.conf does not contain sections.")
+        raise NotImplementedError("softHSM2conf.set method not implemented")
```

Comment:
This is why I asked on the previous comment that we should check if the user provided section. Because we should still be able to add key, value paris to the config file even if it doesn't have sections. If the file doesn't containt sections we can probably just append the key,value pair to the end of the config file.

Reply:
> This is why I asked on the previous comment that we should check if the user provided section. Because we should still be able to add key, value paris to the config file even if it doesn't have sections. If the file doesn't contain sections we can probably just append the key,value pair to the end of the config file.

We use configparser.ConfigParser to handle most of config files. However, config parser works only on config files with sections and it is not able to read or write config files without sections. This splits handling of files with sections from files without sections. It is relevant to check if user provided section (as you suggested) for the files which we expect to have sections(and process with ConfigParser). 
I assume most of the files we will handle will contain sections and therefore methods in File class work with such files while Classes handling files without sections should overwrite these methods.

Reply:
Right, and that is okay. But one more problem that could happen here is that the user provides a legitimate section but the section doesn't exist in the file by default.
For example in sssd.conf we can have a section like  [domain/sc.test.com] but this section won't be there by default and the command will fail.

Reply:
This is in class for softHSM handling. Do we need to add sections to softHSM config file? If not, I will keep this as it is. 
Note, that sssd.conf (and also File class) implements it's own set method that adds sections in case those do not exist.